go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -1249,6 +1254,53 @@ func (call *builtinLastDay) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.Date, Flag: arg.Flag | flagNullable}, nil
 }
 
+func fromDays(loc *time.Location, d int64) *datetime.Date {
+	// 3652424 days corresponds to maximum date i.e. 9999-12-31
+	// mysql returns 0000-00-00 for days below 366
+	if d > 3652424 || d < 366 {
+		return nil
```

Comment:
Since it returns `0000-00-00`, should we here also return a zero date and not `nil`? This also leads to mismatching behavior between MySQL and Vitess now, if I run the tests locally they fail:

```
go test ./go/vt/vtgate/evalengine/integration -run TestMySQL/FnFromDays
--- FAIL: TestMySQL (1.03s)
    --- FAIL: TestMySQL/FnFromDays (0.02s)
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(1) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(255) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(' 0 ') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(' 1 ') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(' 255 ') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('\t1foo\t') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(' 255 foo') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0.0e0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(1.0e0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(1.5e0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(-1.5e0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(1.1e0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(-1.1e0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(-1.7e0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0.0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0.000) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(1.5) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(-1.5) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(1.1) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(1.7) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(-1.1) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(-1.7) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('1.5') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('-1.5') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('foobar') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(_utf8 'foobar') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(_binary 'foobar') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0x0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0x1) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0xff) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(X'00') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(X'01') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(X'ff') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0b1001) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(b'1001') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0x9) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(x'09') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(true) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(false) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0xFF666F6F626172FF) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0x666F6F626172FF) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0xFF666F6F626172) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(9223372036854775807) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(-9223372036854775808) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(18446744073709551615) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(18446744073709540000e0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(JSON_OBJECT()) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(JSON_ARRAY()) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(date '2000-01-01') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(timestamp '2000-01-01 10:34:58') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(timestamp '2000-01-01 10:34:58.123456') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(timestamp '2000-01-01 10:34:58.978654') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(20000101103458) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(20000101103458.1234) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(20000101103458.123456) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(20000101) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('20000101103458') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('20000101103458.1234') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('20000101103458.123456') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('20000101') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('20000101103458foo') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('20000101103458.1234foo') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('20000101103458.123456foo') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('20000101foo') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(time '-10:04:58') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(time '-31:34:58') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(time '-32:34:58') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(time '-101:34:58') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(time '-5 10:34:58') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('10:04:58') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('101:34:58') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS('5 10:34:58') (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(0 as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(1 as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(true as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(false as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('{}' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('[]' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('null' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('true' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('false' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(1 as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(2 as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(1.1 as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(-1.1 as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(9223372036854775807 as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(18446744073709551615 as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('1' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('2' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('1.1' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('-1.1' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('9223372036854775807' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('18446744073709551615' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('"foo"' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('"bar"' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(_binary' "foo"' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(_binary '"bar"' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(0xFF666F6F626172FF as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(0x666F6F626172FF as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(0b01 as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(0b001 as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('["a"]' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('["ab"]' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('["ab", "cd", "ef"]' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('["ab", "ef"]' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('{"a": 1, "b": 2}' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('{"b": 2, "a": 1}' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('{"c": 1, "b": 2}' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast('{"b": 2, "c": 1}' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(date '2000-01-01' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(date '2000-01-02' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(timestamp '2000-01-01 12:34:58' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(time '12:34:56' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(time '12:34:58' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(cast(time '5 12:34:58' as json)) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(0) (SIMPLIFY=false)
            row: []
        comparison_test.go:276: different results: NULL; mysql response: DATE("0000-00-00") (local collation: binary; mysql collation: binary)
            query: SELECT FROM_DAYS(1) (SIMPLIFY=false)
            row: []
```

This indicate Vitess returns the wrong thing here. If we return a zero date, we also don't need a pointer to `datetime.Date` but we can return a `datetime.Date` directly as well if we want to.

Reply:
Oh, fun, we do need to emit a `NULL`, because overflows do result in `NULL` but too small values result in `0000-00-00`. So both can happen and we need to be explicit about it 😞

Reply:
Hmm... I checked. Even values between (3652424, 3652424+76) result in `NULL`

Reply:
Wait, it's even more confusing. There's only a specific range of values where it returns `NULL`. If it's then a much larger number again, it returns the zero date again 😕.

Reply:
```
mysql> SELECT FROM_DAYS(3652424);
+--------------------+
| FROM_DAYS(3652424) |
+--------------------+
| 9999-12-31         |
+--------------------+
1 row in set (0.00 sec)
```

```
mysql> SELECT FROM_DAYS(3652425);
+--------------------+
| FROM_DAYS(3652425) |
+--------------------+
| NULL               |
+--------------------+
1 row in set, 1 warning (0.00 sec)
```

```
mysql> SELECT FROM_DAYS(3652499);
+--------------------+
| FROM_DAYS(3652499) |
+--------------------+
| NULL               |
+--------------------+
1 row in set, 1 warning (0.00 sec)
```

```
mysql> SELECT FROM_DAYS(3652500);
+--------------------+
| FROM_DAYS(3652500) |
+--------------------+
| 0000-00-00         |
+--------------------+
1 row in set (0.00 sec)
```

Reply:
@beingnoble03 Yeah, so we have to special case the values from `3652425` to `3652499` to return `NULL` and then the zero date again for higher values.

Reply:
done.